test_c/test.h

```diff
@@ -3,20 +3,25 @@
 
 #ifdef _WIN32
 
-#ifdef _MSC_VER
-
 #include <stdlib.h>
 
 #ifndef srand48
 #define srand48 srand
 #endif // srand48
 
 #ifndef drand48
-#define drand48() rand() / (RAND_MAX + 1.0)
+extern inline double drand48() {
+    double r = rand() / (RAND_MAX + 1.0);
+    while (r < 0.01) {
+        // BUG:
+        // This library cannot handle input values smaller than 0.01
+        // Increase the value of r until its usable
+        r *= 10.0;
+    }

```

Comment:
I would just call `rand()` again instead of multiplying by 10.

Reply:
actually, disregard that terrible comment

Reply:
Too late for coffee, too. :grin: